commands/roles.js

```diff
@@ -1,24 +1,38 @@
 const Discord = require('discord.js')
 
 exports.run = (client, message, args) => {
-    let roles = message.guild.roles.filter(y => y.name !== '@everyone').array().sort()
+    // Role names inside the blacklist won't be included in the final list
+    // Find all roles below the Coding Yabe Sei role, filter out the blacklist roles, and filter out any roles with the Administrator permission 
+    // (to avoid any no perms message if a user were to try and add the role) - and then sort the filter and convert it into an array
+    try{
+    var blackList = ["@everyone"]
+
+    let yabeRole = message.guild.roles.find(yaberole => yaberole.name == "Coding Yabe Sei")
+    let roles = message.guild.roles.filter(allRoles => allRoles.calculatedPosition < yabeRole.calculatedPosition && !blackList.includes(allRoles.name) 
+    && !allRoles.hasPermission("ADMINISTRATOR"))
+    .sort().array();
+
 
-    let rolesDel = roles.join(' ')
-    let rolesStr = rolesDel.toString()
 
     let embed = new Discord.RichEmbed()
     .setColor(client.config.embedColor)
-    .setTitle(`These are the ${roles.length} roles for ` + message.guild.name + '\n')
-    .setDescription(rolesStr)
-    .addField('Add roles with Yabe with `yabe addrole <role name>`', 'Note: roles higher than Yabe\'s may not be added, so keep all admin roles above Yabe\'s role and all addable roles below Yabe\'s role')
+    .setTitle(`These are the ${roles.length} roles avaliable to you for ` + message.guild.name + '\n')
+    .setDescription(roles.join(' '))
+    .addField('Add roles using Yabe with `yabe addrole <role name>`', 'Note: Only roles below the \`Coding Yabe Sei\` role are displayed')
     .setTimestamp()
 
     message.channel.send(embed)
-    console.log(roles.length)
+    //console.log(roles.length)
+} catch (err) {
+    console.log('There was an error displaying roles: ' + err)
+    }
 }
 
+
 exports.help = {
     name: "roles",
-    description: "The `roles` command displays all roles for the server.",
+    description: "The `roles` command displays all roles below the \`Coding Yabe Sei\` role.",
     usage: "`yabe roles`"
-}
\ No newline at end of file
+}
+
+//Improved by yabe's favorite father-in-law Zen
```

Comment:
everything is fine except for this :squint:

Reply:
It's true - what other father-in-laws does yabe have? :realshit: